packages/core/lib/metadata-delegate.js

```diff
@@ -0,0 +1,53 @@
+module.exports = {
```

Comment:
Unit test would be useful, particularly as documentation to show how it relies on the binding of function context when called. It's not a common pattern to see `this` out of class syntax these days so it's not immediately obvious what `this` is referring to based on this code alone, indeed it depends entirely on how this module is imported/called. So a unit test showing it being used as a sort of class mixin would be useful.

Reply:
Even though this is tested via the client/event tests, do you think we should add more – given we've now switched away from using `this`?

Reply:
Ah yeah, I didn't click on it was being covered as part of integration tests.